### PR TITLE
Update workflows

### PR DESCRIPTION
**Pull request by torredil**
**What is this PR about? / Why do we need it?**
- Add workflow to scan container images (CSI driver + sidecars) when changes are made in `'deploy/kubernetes/overlays/stable/gcr/kustomization.yaml'` and daily.
- Add workflow to scan on repo mode which helps identify potential vulnerabilities that might get introduced with new PRs.
- Remove `codeQL` workflow.

_Signed-off-by: Eddie Torres <torredil@amazon.com>_

### CODE REVIEW

File: .github/workflows/trivy-sidecars.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Trivy Scanner - Container Images
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'deploy/kubernetes/overlays/stable/gcr/kustomization.yaml'
+  schedule:
+   # Run “At minute 0 past every 24th hour.”
```
Comment (ConnorJC3) [inline .github/workflows/trivy-sidecars.yaml:9]: Remove this comment

File: .github/workflows/trivy.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Run Trivy vulnerability scanner in repo mode
+        uses: aquasecurity/trivy-action@master
+        with:
+          scan-type: 'fs'
+          ignore-unfixed: true
+          format: 'sarif'
+          output: 'trivy-ebs-csi-results.sarif'
```
Comment (ConnorJC3) [inline .github/workflows/trivy.yaml:19]: Probably just call this `results.sarif` to be consistent with the other workflow?

File: .github/workflows/trivy-sidecars.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+name: Trivy Scanner - Container Images
```
Comment (ConnorJC3) [inline .github/workflows/trivy-sidecars.yaml:1]: I think this file should be renamed to `trivy-containers.yaml` or similar now that we're not just scanning the sidecars

File: .github/workflows/trivy.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+name: Trivy Scanner - EBS CSI Driver
```
Comment (ConnorJC3) [inline .github/workflows/trivy.yaml:1]: Change name to "Trivy Scanner"

**Review by ConnorJC3 — CHANGES_REQUESTED** [review]

File: .github/workflows/trivy.yaml (hunk 4 above)
Comment (torredil) [reply .github/workflows/trivy.yaml]: Done.

**Pull request merged**